larisa.py

```diff
@@ -8,59 +8,82 @@
 import requests
 from geopy import Nominatim
 
-def weather_forecast(city):
+
+def temperature_conversion(temp):
+    '''
+    #Author: Nils Becker
+    Converts a given temperature in °K into a °C String representation
+    :param temp: Float Value of temperature in °K
+    :returns: The temperature converted to °C as a String
+    '''
+    return f"{int(temp)-273}°C"
+
+def weather_forecast(cfg, city):
     '''
     #Author: Nils Becker
     Gives a weather forecast for a specified city
+    :param cfg: Currently used configuration
     :param city: String representation of city
     :returns: weather forecast for now/ next 12h/ next 7 days in a List
     '''
     locator = Nominatim(user_agent="Larissa")
     location = locator.geocode(city)
-    api_key = '14ebab67cd0768d48dd1eb7545d89ac1'
-    url = 'https://api.openweathermap.org/data/2.5/onecall?lat={}&lon={}&exclude={}&appid={}'.format(location.latitude, location.longitude, "minutely,alerts", api_key)
-    response = requests.get(url).json()
+    api_key = cfg['weather']['apikey']
+    try:
+    #create API request for given city
+        url = 'https://api.openweathermap.org/data/2.5/onecall?lat={}&lon={}&exclude={}&appid={}'.format(location.latitude, location.longitude, "minutely,alerts", api_key)

```

Comment:
Just to check, maybe URL can also become a part of a config?

Reply:
I thought about it as well but as subtantial parts of the URL consist of city dependend (and by that input depenend), I decided against it.